Initial Welcome page

timtmok opened the pull request:
### Intent
Address #2816

### Approach
The eslint config was updated to ignore a react-hooks rule for `.ts` files.

Removes the Walkthrough section and moves the Open and Recent sections to the right. The left side has a React container injected to hold the Start and Help sections.

New Notebook - This shows a modal popup to select which interpreter to use. The new notebook command has an optional argument to set the interpreter.

New File - Invokes the new file command. The user interacts with the command palette to select the file type.

New Console - This re-uses the select interpreter dropdown in the top action bar. Some parts may need customization (power up button doesn't fit here). The console pane is maximized as well.

New Project - Invokes the new project wizard command.

The Help section is a set of links. The Github repo link has been added to the trusted domains to avoid the prompt about a trusted link.

The Open section also has some links removed that were redundant or should be hidden until we decide to support them.

![image](https://github.com/posit-dev/positron/assets/9591545/f13301b5-586b-4bef-9168-fde3d56650c8)


### QA Notes
See [spec](https://docs.google.com/document/d/1sxYale5_i31MaRooOxq2DeovZiNmy_-nGlTxbqbVNT0/edit?usp=sharing)

A flag still needs to be added to be able to test in release builds

timtmok committed:
Re-organize Welcome page

Remove Walkthroughs section
Move Open and Recent sections to the right
Inject React for Positron specific sections in the left column

timtmok committed:
Customize Welcome page

Injects a React container for Start and Help sections
Removes Walkthroughs
Adds logo and themed product name

sharon-wang reviewed (commented):
Great to see the initial welcome page! 🎉 

Overall LGTM, just some colour contrast and layout comments.

timtmok committed:
Adjust appearance

Fix contrast for logo and text
Make Start buttons equal width
Add border radius to buttons for high contrast theme
Increase max width for welcome container

sharon-wang reviewed (approved):
Stamping the checkmark ✅ Any more-involved enhancements can be added separately as needed!
